docs/setup.md

```diff
@@ -50,16 +64,93 @@ To get help regarding the usage of service account setup command within the snap
+* ***primary*** - A marker to indicate the current service account should be made 'primary' for implicit spark-submit job submission purposes.
+* ***properties-file*** - File with all configuration properties to be associated with a service account.
+* ***conf*** - Values to add to and override the ones in specified properties-file param.
 
 ```bash
-usage: setup-spark-k8s.py service-account [-h] [--username USERNAME] [--namespace NAMESPACE]
+usage: setup-spark-k8s.py service-account [-h] [--primary] [--properties-file PROPERTIES_FILE] [--conf CONF]
 
 optional arguments:
   -h, --help            show this help message and exit
-  --username USERNAME   Service account username to be created in kubernetes. Default is spark
-  --namespace NAMESPACE
-                        Namespace for the service account to be created in kubernetes. Default is default namespace
+  --primary             Boolean to mark the service account as primary.
+  --properties-file PROPERTIES_FILE
+                        File with all configuration properties assignments.
+  --conf CONF           Config properties to be added to the service account.
+```
+Service account is an abstraction for a set of associated kubernetes resources needed to run a Spark job. The user can choose to associate configuration properties 
+with the service account that can serve as default while submitting jobs against that service account from any machine within the kubernetes cluster. A typical use 
+of this feature would look like this.
+
+```bash
+spark-client.setup-spark-k8s --username demouser --namespace demonamespace service-account --properties-file /home/demouser/conf/spark-defaults.conf --conf spark.app.name=demo-spark-app --conf spark.executor.instances=3
+```
+
+The above command sets up a service account for user ```demonamespace:demouser``` for Spark job submission using configuration properties coming from the specified 
+properties file while overriding the configuration properties ```spark.app.name``` and ```spark.executor.instances```.
+
+As you will discover further in the documentation for [job submission](/docs/submit.md), this service account along with it's default configuration properties can be used to submit a Spark job.
+
+For example, assuming the properties file provided has configuration details to access data in S3, one could submit a job like
+```bash
+spark-client.spark-submit  --username demouser --namespace demonamespace --deploy-mode cluster --conf spark.app.name=demo-spark-s3-app $S3_PATH_FOR_CODE_FILE
+```
+This would launch the spark job with configuration coming from the service account for user ```demonamespace:demouser``` but the app name would be ```demo-spark-s3-app```. 
+
+**_NOTE:_** The comamnd described above does not create a kubernetes namespace but needs it to be there. It does however create the requested username in the specified and existing namespace. 
+
+#### Creating configuration for a Service Account after account creation
```

Comment:
These look more like a "How to Guide". I would probably just make a section called "How To" , where we put these commands

Reply:
Done